Imroved Settings page, Updated content.js coupons auto-apply.

Abstra208 opened the pull request.

Abstra208 committed:
refactor: simplify UI components and improve language/theme switchers

Abstra208 committed:
refactor: update coupon fetching logic to use localStorage for domain and coupons

Abstra208 committed:
Merge branch 'Abdallah-Alwarawreh:main' into main

Abstra208 committed:
feat: add CustomDatabase component and integrate it into Settings

Abstra208 committed:
feat: enhance CustomDatabase component with URL validation and localization support

Abstra208 committed:
Merge branch 'Abdallah-Alwarawreh:main' into main

Abstra208 committed:
feat: improve coupon fetching and storage logic, add support for fossil.com

Abstra208 committed:
fix: correct coupon iteration logic in content.js

Abstra208 committed:
feat: add coupon data for Fossil and enhance coupon handling logic

Abstra208 committed:
Merge branch 'Abdallah-Alwarawreh:main' into main

Abdallah-Alwarawreh commented on Extension-React/coupons.json:
```diff
@@ -0,0 +1,146 @@
+{
```
what is even this file?

Abstra208 replied:
I forgot to remove it.

Abdallah-Alwarawreh replied:
oh lmao

Abstra208 committed:
chore: remove outdated coupon data from coupons.json

ImGajeed76 commented on Extension-React/public/background.js:
```diff
@@ -1,12 +1,26 @@
-                request.domain
-        )
+        let database = "";
+        try {
+            if (localStorage.getItem("database") === null) {
+                database = "https://api.discountdb.ch/api/v1/syrup/coupons";
+            } else {
+                database = localStorage.getItem("database");
+            }
+        } catch (e) {
+            database = "https://api.discountdb.ch/api/v1/syrup/coupons";
+        }
+        fetch(database + "?domain=" + request.domain)
             .then((response) => response.json())
             .then((data) => {
-                sendResponse({ coupons: data });
+                chrome.storage.local.set({ coupons: data }, () => {
+                    if (chrome.runtime.lastError) {
+                        console.error("Error setting coupons:", chrome.runtime.lastError);
+                        sendResponse({ coupons: [] });
+                    } else {
+                        sendResponse({ coupons: data });
+                    }
+                });
```
Please use the types/classes/functions from the sas client. I know you said thats not possible but as of a quick search it should be. Though i didnt try yet.

Abstra208 replied:
I cannot import anything in content.js

ImGajeed76 reviewed (commented):
I will have a look at the rest later.

black-backdoor commented on Extension-React/public/content.js:
```diff
@@ -844,7 +860,8 @@
         await logger.init();
-        let domain = window.location.hostname.replace("www.", "");
+        let domain = window.location.hostname.replace("www.", "").replace("https://", "").replace("http://", "");
+        console.error(domain);
```
Using `console.error(domain)` might be misleading

Abstra208 replied:
It's just a debug message

black-backdoor reviewed (commented):
While testing the popup for coupons, it currently displays:
![screenshot](https://github.com/user-attachments/assets/ba1a1c76-a38e-4ecc-bd92-beceb3e1909c)

Abstra208 committed:
chore: remove debug logging from content.js

Abstra208 reviewed (commented):
> While testing the popup for coupons, it currently displays:
> ![screenshot](https://github.com/user-attachments/assets/ba1a1c76-a38e-4ecc-bd92-beceb3e1909c)
>  

Are you using firefox?
What is your browser's languages set to?

black-backdoor commented:
> Are you using firefox? What is your browser's languages set to?

I tried manually setting the language to several different options.

Previously, I didn't have much time, so I couldn't run detailed tests, but I encountered a error
`[Syrup] Failed to load translations: TypeError: Failed to fetch` 

I also tried the same approach with the code from the main branch, but I encountered the same error.


EDIT: it's trying to load `/_locales/en-GB/messages.json` #107

Abstra208 committed:
feat: enhance message listener and add fallback for translation loading

Abstra208 committed:
Merge branch 'Abdallah-Alwarawreh:main' into main

Abstra208 committed:
Added back package-lock.json

They were removed in #113, but were needed for the website and the extension to build .

Abdallah-Alwarawreh reviewed (approved):
Lgtm
